File: src/backend/frotas/src/model/frota_veiculo_pneu_model.py
from src import db


class FrotaVeiculoPneuModel(db.Model):
    __tablename__ = 'frota_veiculo_pneu'

    id = db.Column(db.Integer, primary_key=True)
    data_troca = db.Column(db.DateTime)
    valor_troca = db.Column(db.Float)
    posicao_pneu = db.Column(db.String(100))
    marca_pneu = db.Column(db.String(100))
    id_frota_veiculo = db.Column(db.Integer, db.ForeignKey('frota_veiculo.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_frota_veiculo = data.get('idFrotaVeiculo')
        self.data_troca = data.get('dataTroca')
        self.valor_troca = data.get('valorTroca')
        self.posicao_pneu = data.get('posicaoPneu')
        self.marca_pneu = data.get('marcaPneu')

    def serialize(self):
        return {
            'id': self.id,
            'idFrotaVeiculo': self.id_frota_veiculo,
            'dataTroca': self.data_troca.isoformat(),
            'valorTroca': self.valor_troca,
            'posicaoPneu': self.posicao_pneu,
            'marcaPneu': self.marca_pneu,
        }

File: frontend/frotas/lib/app/data/model/frota_veiculo_pneu_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:frotas/app/infra/infra_imports.dart';
import 'package:intl/intl.dart';

class FrotaVeiculoPneuModel {
	int? id;
	int? idFrotaVeiculo;
	DateTime? dataTroca;
	double? valorTroca;
	String? posicaoPneu;
	String? marcaPneu;

	FrotaVeiculoPneuModel({
		this.id,
		this.idFrotaVeiculo,
		this.dataTroca,
		this.valorTroca,
		this.posicaoPneu,
		this.marcaPneu,
	});

	static List<String> dbColumns = <String>[
		'id',
		'data_troca',
		'valor_troca',
		'posicao_pneu',
		'marca_pneu',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Data Troca',
		'Valor Troca',
		'Posicao Pneu',
		'Marca Pneu',
	];

	FrotaVeiculoPneuModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idFrotaVeiculo = jsonData['idFrotaVeiculo'];
		dataTroca = jsonData['dataTroca'] != null ? DateTime.tryParse(jsonData['dataTroca']) : null;
		valorTroca = jsonData['valorTroca']?.toDouble();
		posicaoPneu = jsonData['posicaoPneu'];
		marcaPneu = jsonData['marcaPneu'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idFrotaVeiculo'] = idFrotaVeiculo != 0 ? idFrotaVeiculo : null;
		jsonData['dataTroca'] = dataTroca != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataTroca!) : null;
		jsonData['valorTroca'] = valorTroca;
		jsonData['posicaoPneu'] = posicaoPneu;
		jsonData['marcaPneu'] = marcaPneu;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idFrotaVeiculo = plutoRow.cells['idFrotaVeiculo']?.value;
		dataTroca = Util.stringToDate(plutoRow.cells['dataTroca']?.value);
		valorTroca = plutoRow.cells['valorTroca']?.value?.toDouble();
		posicaoPneu = plutoRow.cells['posicaoPneu']?.value;
		marcaPneu = plutoRow.cells['marcaPneu']?.value;
	}	

	FrotaVeiculoPneuModel clone() {
		return FrotaVeiculoPneuModel(
			id: id,
			idFrotaVeiculo: idFrotaVeiculo,
			dataTroca: dataTroca,
			valorTroca: valorTroca,
			posicaoPneu: posicaoPneu,
			marcaPneu: marcaPneu,
		);			
	}

	
}